lib/Runtime/Library/BuiltIn/IndexOf.js

```diff
@@ -38,7 +38,7 @@
     }
 
     while (k < len) {
-        if (o[k] === searchElement && (o[k] !== undefined || searchElement !== undefined)) {
+        if (o[k] === searchElement && (o[k] !== undefined || (searchElement === undefined && o.hasOwnProperty(k)))) {
```

Comment:
I think you need to say `k in o`. Likely now we are going to incorrectly skip case where `o.prototype[k]` is `undefined`.